test/active_record_cached_test.rb

```diff
@@ -0,0 +1,141 @@
+require_relative './test_helper'
+
+class I18nBackendActiveRecordCachedTest < I18n::TestCase
+  def setup
+    I18n.backend = I18n::Backend::ActiveRecord.new
+    store_translations(:en, :foo => { :bar => 'bar', :baz => 'baz' })
+
+    I18n::Backend::ActiveRecord.configure do |config|
+      config.cache_translations = true
+    end
+  end
+
+  def teardown
+    I18n::Backend::ActiveRecord::Translation.destroy_all
+    I18n::Backend::ActiveRecord.instance_variable_set :@config, I18n::Backend::ActiveRecord::Configuration.new
+    super
+  end
+
+  test "store_translations does not allow ambiguous keys (1)" do
```

Comment:
Just thinking how to avoid duplicating tests
Maybe we could the following
```ruby
[false, true].each do |flag|
  test "store_translations does not allow ambiguous keys when #{'not ' unless flag}cache_translations (1)" do
    I18n::Backend::ActiveRecord.configure { |config| config.cache_translations = flag }
    I18n::Backend::ActiveRecord::Translation.delete_all
    I18n.backend.store_translations(:en, :foo => 'foo')
    # ...
  end
end
```
I just really want to avoid duplicating tests

Reply:
We could also try
```ruby
class I18nBackendActiveRecordCachedTest < I18n::TestCase
  # ...

  [false, true].each do |flag|
    Class.new I18n::TestCase do
      def setup
        I18n::Backend::ActiveRecord.configure { |config| config.cache_translations = flag }
      end

      # or maybe
      # define_method :setup do
      #   I18n::Backend::ActiveRecord.configure { |config| config.cache_translations = flag }
      # end

      test "store_translations does not allow ambiguous keys (1)" do
        # ...
      end

      test "store_translations does not allow ambiguous keys (2)" do
        # ...
      end

      # ...
    end
  end

  # ...
end
```
Basically emulate the context gem, but I am not entirely sure it would work

Reply:
Ended up going with the first idea, since the second option didnt work well the first time.

This resulted in a fully passing spec :)

Reply:
Went with the first option because the second option didnt work well the first time.

We now have a consistent test suite. Its all passing except has 6 failures, 3 per flag setting, regarding #106